util/reggen/register.py

```diff
@@ -95,6 +104,8 @@ def __init__(self,
         super().__init__(offset)
         self.name = name
         self.desc = desc
+        self.async_name = async_name
+        self.async_clk = async_clk
```

Comment:
Could we pass the clocking information to the Register constructor so that we can do the lookup here and get rid of the `handle_async` function? That way, we avoid having a partially constructed object that needs "finishing off" later.

Reply:
yep! this one and the two above i'm doing together. It actually does remove lots of code.